src/analysis/destruct.ml

```diff
@@ -571,9 +590,9 @@ let rec node config source selected_node parents =
             | [ more_precise ] ->
               (* If only one pattern is generated, then we're only refining the
                 current pattern, not generating new branches. *)
+              let punning_field = find_field_name_for_punning_field patt parents in
```

Comment:
I think I would prefer something like `punned_field` here, but that's not really important.